crypto/src/action/spend.rs

```diff
@@ -0,0 +1,80 @@
+use ark_ff::UniformRand;
+use rand_core::{CryptoRng, RngCore};
+
+use crate::Fr;
+
+use crate::{
+    keys::SpendKey,
+    merkle::proof::MerklePath,
+    proofs::transparent::SpendProof,
+    rdsa::{Signature, SigningKey, SpendAuth, VerificationKey},
+    value, Note, Nullifier,
+};
+
+pub struct Spend {
+    pub body: Body,
+    pub auth_sig: Signature<SpendAuth>,
+}
+
+impl Spend {
+    pub fn new<R: RngCore + CryptoRng>(
+        mut rng: R,
+        spend_key: SpendKey,
+        _merkle_path: MerklePath,
+        note: Note,
+    ) -> Self {
+        // TODO: Derive nullifier from note commitment, note position, and
+        // nullifier deriving key
+        // See p.55 ZCash spec
+        let nullifier = Nullifier::new();
+
+        let v_blinding = Fr::rand(&mut rng);
```

Comment:
I think we're going to need to hold on to this value to be able to construct binding signatures.